lib/galaxy/tools/__init__.py

```diff
@@ -1079,6 +1080,24 @@ def _parse_citations(self, tool_source):
                     citations.append(citation)
         return citations
 
+    def _parse_xrefs(self, tool_source):
+        if not hasattr(tool_source, 'root'):
+            return []
+        root = tool_source.root
+        xrefs = []
+        xrefs_elem = root.find("xrefs")
+        if xrefs_elem is None:
+            return xrefs
+
+        for xref_elem in xrefs_elem:
+            if xref_elem.tag != "xref":
+                pass
+            if hasattr(self.app, 'xrefs_manager'):
```

Comment:
Not sure we need a manager for xrefs, probably it's enough to parse `xref_elem` directly here.

Reply:
Something like that should wok instead?
```
for xref_elem in xrefs_elem:
            if xref_elem.tag != "xref":
                pass
            xref = dict(
                reftype=xref_elem.text.strip(),
                content=xref_elem.attrib.get('type', None)
            )
            if xref:
                xrefs.append(xref) 
```

Reply:
Almost correct:
```python
        for xref_elem in xrefs_elem:
            if xref_elem.tag != "xref":
                pass
            xref = dict(
                content=xref_elem.text.strip(),
                reftype=xref_elem.attrib.get('type', None)
            )
            xrefs.append(xref)
```

Reply:
https://github.com/galaxyproject/galaxy/pull/7905#issuecomment-505405772